lib/resty/etcd/v3.lua

```diff
@@ -29,8 +32,57 @@ local mt = { __index = _M }
 
 -- define local refresh function variable
 local refresh_jwt_token
+local fails
 
-local function _request_uri(self, method, uri, opts, timeout, ignore_auth)
+local function fault_count(key, shm_name, fail_timeout)
+    local new_value, err, forcible = ngx_shared[shm_name]:incr(key, 1, 0, fail_timeout)
+    if err then
+        return nil, err
+    end
+
+    if forcible then
+        utils.log_warn("shared dict: ", shm_name, " is full, valid items forcibly overwritten")
+    end
+    return new_value, nil
+end
+
+
+local function restore(fail_timeout, endpoint)
+    local ok, err = ngx_timer_at(fail_timeout, function(premature)
+        if premature then
+            return
+        end
+
+        utils.log_info("restore an endpoint to health: ", endpoint.http_host)
+        endpoint.health_status = 1
```

Comment:
Better to use warn log when we change the health_status. It could be helpful when we need to do accounting.